### PR TITLE
BHoM_Engine: Add nuspec template file

### LINKED ISSUE #3105: Add a nuspec file
To the folder `.ci/BHoMBot/Nuget` - in the form of `{repository_name}.nuspec` to help the bot automatically produce nugets of this repository.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Fixes #3105

### CODE REVIEW

**Review by Dhorix — APPROVED** [review]
This looks good, okay to to commit

File: .ci/BHoMBot/Nuget/BHoM.Engine.nuspec [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      <group targetFramework="netstandard2.0">
+        <dependency id="NETStandard.Library" version="2.0.3" />
+        <dependency id="Microsoft.Windows.Compatibility" version="5.0.2" />
+        <dependency id="Microsoft.CSharp" version="4.7.0" />
+        <dependency id="NETStandard.Library" version="2.0.3" />
+        <dependency id="MongoDB.Bson" version="2.18.0" />
+        <dependency id="Mono.Reflection" version="2.0.0" />
+        <dependency id="CompareNETObjects" version="4.67.0" />
+      </group>
+      <group targetFramework=".NETFramework4.5.2">
```
Comment (FraserGreenroyd) [inline .ci/BHoMBot/Nuget/BHoM.Engine.nuspec:27]:
```suggestion
      <group targetFramework="net472">
```

**Review by kprusicka — APPROVED** [review]
Looks amazing to me

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: code-compliance, copyright-compliance, core, documentation-compliance, installer, project-compliance, ready-to-merge, serialisation, versioning, dataset-compliance, unit-tests

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: null-handling

**Pull request merged**